readthedocs/proxito/views/hosting.py

```diff
@@ -657,47 +658,16 @@ def _get_filetreediff_response(self, *, request, project, version, resolver):
         if not diff:
             return None
 
-        return {
-            "enabled": True,
-            "outdated": diff.outdated,
-            "diff": {
-                "added": [
-                    {
-                        "filename": filename,
-                        "urls": {
-                            "current": resolver.resolve_version(
-                                project=project,
-                                filename=filename,
-                                version=version,
-                            ),
-                            "base": resolver.resolve_version(
-                                project=project,
-                                filename=filename,
-                                version=base_version,
-                            ),
-                        },
-                    }
-                    for filename in diff.added
-                ],
-                "deleted": [
-                    {
-                        "filename": filename,
-                        "urls": {
-                            "current": resolver.resolve_version(
-                                project=project,
-                                filename=filename,
-                                version=version,
-                            ),
-                            "base": resolver.resolve_version(
-                                project=project,
-                                filename=filename,
-                                version=base_version,
-                            ),
-                        },
-                    }
-                    for filename in diff.deleted
-                ],
-                "modified": [
+        def _filter_diff_files(files):
+            # Filter out all the files that match the ignored patterns
+            for filename in files.copy():
+                for ignore_pattern in project.addons.filetreediff_ignored_files or []:
+                    if fnmatch.fnmatch(filename, ignore_pattern):
+                        files.remove(filename)
```

Comment:
You can write this as list comprehension, so you can avoid doing in-place operations.

```suggestion
            files = [
                filename
                for filename in files
                if not any(
                    fnmatch.fnmatch(filename, ignore_pattern)
                    for ignore_pattern in ignore_patterns
                )
            ]
```